lib/plaid/user.rb

```diff
@@ -112,12 +166,29 @@ def transactions(pending: false, account_id: nil, start_date: nil, end_date: nil
     #
     # Returns self.
     def update(username, password, pin = nil)
```

Comment:
one thing that was a little confusing to me about understanding the `User` class is that we are storing a lot of data state and also performing stateful actions (server requests). It's a little tough to reason about the correctness of the data model as various updates happen. 

One thing I'd like to keep very clear to our customers is which things represent a stateful server call from which things are purely client-side data. I'm wondering if we should separate things more and have separate classes for modeling data vs. performing stateful actions. I'm not 100% sure on this as the Rails way is to keep model data on the object and also to allow it to perform state changes (ex ActiveRecord's `save` method) and sometimes folks in the Ruby community expect to see that design rather than what I'm proposing. What do you think?